modules/apps/frontend-view-state/frontend-view-state-service/src/main/java/com/liferay/frontend/view/state/service/impl/FVSFrontendDataSetEntryLocalServiceImpl.java

```diff
@@ -28,4 +35,42 @@
 )
 public class FVSFrontendDataSetEntryLocalServiceImpl
 	extends FVSFrontendDataSetEntryLocalServiceBaseImpl {
+
+	@Override
+	public FVSFrontendDataSetEntry addFVSFrontendDataSetEntry(
+			String fdsName, long fvsEntryId, String name, long plid,
```

Comment:
`name` is the name of the custom view. A user sets this in UI (see modal in gif). It is basically a label. Unique identifier of a custom view is `fvsFrontendDataSetEntryId`. 

`fdsName` is the unique identifier of the FDS instance. It is set as the `id` attribute / prop of FDS, for example [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/customized.jsp#L34). I am not sure if we should call it `fdsId` or `fdsName`. For me, it makes sense to call a descriptive string identifier a `name`, and non-descriptive numeric identifier `id`. I think that Brian also feels that way, since he introduced the convention to have FDS identifiers in `*FDSNames` constant files.

By 'external identifier', do you mean primary key of some other DB table? FDS instance, along with its identifier, is not persisted. It is defined in FDS tag or component configuration, exists only in code.